R/modify_teal_module_report.R

```diff
@@ -0,0 +1,178 @@
+                         ggplot2_args,
+                         default_outlier_label,
+                         decorators)
+
+    moduleServer(id, function(input, output, session) {
+
+      original_outputs <- original_server(
+        'modified',
+        data,
+        filter_panel_api,
+        response = response,
+        regressor = regressor,
+        reporter,
+        plot_height,
+        plot_width,
+        ggplot2_args,
+        default_outlier_label,
+        decorators
+      )
+      # work on original_outputs
+    })
+
+  module(
+    label = teal_module$label,
+    server = new_server,
+    ui = new_ui,
+    datanames = teal_module$datanames,
+    server_args = teal_module$server_args,
+    ui_args = teal_module$ui_args,
+    transformators = teal_module$transformators
+  )
+}
+
+
+modify_teal_module_output <- function(teal_module, modify_fun, output_name) {
+  stopifnot(inherits(teal_module, "teal_module"))
+
+  original_server <- teal_module$server
+  original_ui <- teal_module$ui
+  new_ui <- function(id, ...){
+    ns <- NS(id)
+    original_ui(ns('modified'), ...)
+  }
+
+  original_server <- teal_module$server
+  new_server <- function(...) { }
+  formals(new_server) <- formals(original_server)
+
+  body(new_server) <- quote({
+
+    arg_names <- names(formals(original_server))
+
+    # Capture current call and evaluate each argument
+    call_expr <- match.call()
+    call_args <- as.list(call_expr)[-1]  # remove function name
+    call_args <- call_args[names(call_args) %in% arg_names]
+
+    # Evaluate all arguments in the current environment
+    eval_args <- lapply(call_args, eval, envir = parent.frame())
+
+    moduleServer(id, function(input, output, session) {
+      eval_args$id <- 'modified'
+      original_outputs <- do.call(original_server, eval_args)
+      if (!output_name %in% names(original_outputs)) {
+        stop(paste0("The provided teal_module does not return ", output_name))
+      }
+      stopifnot(is.reactive(original_outputs[[output_name]]))
+
+      modified_output <- reactive({
+        modify_fun(original_outputs[[output_name]]())
+      })
+
+      named_modified_output <- setNames(list(modified_output), output_name)
+      return(modifyList(original_outputs, named_modified_output))
+
+    })
+  })
+
+  module(
+    label = teal_module$label,
+    server = new_server,
+    ui = new_ui,
+    datanames = teal_module$datanames,
+    server_args = teal_module$server_args,
+    ui_args = teal_module$ui_args,
+    transformators = teal_module$transformators
+  )
+}
+
+modify_teal_module_report_card <- function(teal_module, modify_fun) {
+  modify_teal_module_output(teal_module, modify_fun, output_name = 'report_card')
+}
+
+nullify_teal_module_report_card <- function(teal_module) {
+  modify_teal_module_output(teal_module, modify_fun = function(report_card) NULL, output_name = 'report_card')
+}
+
+disable_teal_module_report <- nullify_teal_module_report_card
```

Comment:
Redundant code and name could be better.
```suggestion
disable_report <- function(teal_module) {
  checkmate::assert_class(teal_module, "teal_module")
  modify_teal_module_output(teal_module, modify_fun = function(report_card) NULL, output_name = 'report_card')
}
```